asmcomp/asmlink.ml

```diff
@@ -203,6 +203,9 @@ let scan_file obj_name tolink = match read_file obj_name with
 
 (* Second pass: generate the startup file and link it with everything else *)
 
+let force_linking_of_startup ppf =
+  Asmgen.compile_phrase ppf (Cmm.Cdata([Cmm.Cglobal_symbol "caml_startup"]))
+
```

Comment:
Like @shindere I'm tempted to have this (or maybe its two calls) protected by `if !Clflags.output_complete_object then ...`, if only to document-in-code that this forced linking is required only in that case.